lib/mongo/auth/user.rb

```diff
@@ -154,6 +156,12 @@ def initialize(options)
       def spec
         { pwd: hashed_password, roles: roles }
       end
+
+      private
+
+      # Sets the client key for the user.
+      # @return [ String ] The client key for the user.
+      attr_accessor :client_key
```

Comment:
you can make this ``attr_reader`` if you want, and then the line above ``user.instance_variable_set(:@client_key, client_key)`` to avoid the awkwardness of ``user.send(:client_key=, @client_key)``